### DISCUSSION TITLE
FILENAME googlebooks

### DISCUSSION THREAD

**Discussion by Brobin16** [CollectionBuilder/collectionbuilder.github.io#134] [Q&A]
Is it possible to integrate Google Books URLs into a CollectionBuilder project? I attempted this, but encountered loading issues, indicated by spinning wheels. Are there known workarounds or alternative approaches to incorporate Google Books content in CollectionBuilder?

**Comment by evanwill** [comment]
@Brobin16 I am not aware of any CB project currently integrating a Google Books embed. 

There is a [Google Books embedded viewer API](https://developers.google.com/books/docs/viewer/developers_guide) which in theory could be used to add google books content to a CB Item page--I haven't used it, but it doesn't look too complicated! You can't just use the standard book share link, it seems to need an identifier for the book to feed to the embed--so you would probably add that id to the object_location or a new custom column. 

Alternatively, out of the box you could just link out to the Google Book location (rather than embedding)--put the google book link in the object_location and use display_template "record". You could take it a bit further by creating a custom display_template layout that would give you more control over the look (for example, displaying a cover image, but then having a "View on Google Books" button for the link). 

Share a link to your repository if you would like me to take a look. Good luck!
